astroid/nodes/node_classes.py

```diff
@@ -5389,13 +5413,13 @@ def const_factory(value: Any) -> List | Set | Tuple | Dict | Const | EmptyNode:
         node.object = value
         return node
 
-    # TODO: We pass an empty list as elements for a sequence
-    # or a mapping, in order to avoid transforming
-    # each element to an AST. This is fixed in 2.0
-    # and this approach is a temporary hack.
     initializer_cls = CONST_CLS[value.__class__]
```

Comment:
```suggestion
    instance: List | Set | Tuple | Dict
    initializer_cls = CONST_CLS[value.__class__]
```
The type definition for `instance` helps avoid mypy errors in the second `if` block. Without it `instance` would be limited to `List | Set | Tuple`.

Reply:
Ah, `pyright`  thought this was okay so I didn't notice.